### PR TITLE
Added a linting job that triggers on PR events

### PR DESCRIPTION
**Pull request by MitchellBerend**

### CODE REVIEW

**Comment by MitchellBerend** [comment]
Maybe you should also add branch protections that requires this job to pass so you don't shoot yourself in the foot as well. There is also a setting in there somewhere that only allows squash merges so the git history doesn't look like shit.

File: cmd/program/program.go [hunk 1]
```diff
@@ -46,8 +46,11 @@ var (
 func (p *Project) ExitCLI(tprogram *tea.Program) {
 	if p.Exit {
 		// logo render here
-		tprogram.ReleaseTerminal()
-		os.Exit(1)
+		if err := tprogram.ReleaseTerminal(); err != nil {
+			log.Fatal(err)
+		} else {
+			os.Exit(1)
+		}
```
Comment (Melkeydev) [inline cmd/program/program.go:53]: I dont think this if /else is necessary

File: .github/workflows/linting.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.21.1'
+
+      - name: Install dependencies
+        run: go mod tidy
+
+      - name: Install dependencies
+        run: go mod download
```
Comment (Melkeydev) [inline .github/workflows/linting.yml:21]: Two install dependencies steps?

File: .github/workflows/linting.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+name: Linting
```
Comment (Melkeydev) [inline .github/workflows/linting.yml:1]: I am wondering if this can be a pre-commit step? Just asking if this is necessary step to do here

**Review by Melkeydev — CHANGES_REQUESTED** [review]
Wondering about the execution of this Job and if we really need it

**Comment by MitchellBerend** [comment]
This job isn't strictly necessary as a precommit script can do the same, but that could be skipped or forgotten. It's on the maintainer to install the script, I think having a safety net for that makes sense.

**Review by Melkeydev — APPROVED** [review]
LGTM

**Pull request merged**